jvm/scalactic/src/main/scala/org/scalactic/Prettifier.scala

```diff
@@ -277,6 +277,8 @@ private[scalactic] class TruncatingPrettifier(sizeLimit: SizeLimit) extends Defa
         else
           theToString
       // SKIP-SCALATESTJS,NATIVE-END
+      case caseClazz: Product =>
+        s"${caseClazz.productPrefix}(" + caseClazz.productIterator.map(prettify(_, processed + caseClazz.productIterator)).mkString(", ") + ")"
```

Comment:
First of all thanks for the PR!  To be consistent, I think productIterator here should follow the size limit, and the processed part should be processed + caseClazz.  Also, I think it will be good idea to add this to the DefaultPrettifier also.  :)

Reply:
> First of all thanks for the PR! To be consistent, I think productIterator here should follow the size limit

Are you sure about this? There can be un-intended consequences because the ordering is based on the how the fields are defined in the case class. Also the amount of fields in a case case class is typically are not that large (as opposed to the actual data in the case class). Finally this behaviour would be quite surprising for me, as a user I wouldn't expect truncation to occur on the fields themselves, just the data,

> and the processed part should be processed + caseClazz

Nice catch

> Also, I think it will be good idea to add this to the DefaultPrettifier also. :)

Sure will do, its trivial to do so.

Reply:
> > First of all thanks for the PR! To be consistent, I think productIterator here should follow the size limit
> 
> Are you sure about this? There can be un-intended consequences because the ordering is based on the how the fields are defined in the case class. Also the amount of fields in a case case class is typically are not that large (as opposed to the actual data in the case class). Finally this behaviour would be quite surprising for me, as a user I wouldn't expect truncation to occur on the fields themselves, just the data.
> 
> > and the processed part should be processed + caseClazz
> 
> Nice catch
> 
> > Also, I think it will be good idea to add this to the DefaultPrettifier also. :)
> 
> Sure will do, its trivial to do so.

Hmm, that's true.  It is just that it seems in-consistent with other guys in the room, but I think your justification makes good sense.  :)